tapdb/interfaces.go

```diff
@@ -113,7 +115,20 @@ func (t *TransactionExecutor[Q]) ExecTx(ctx context.Context,
 	}()
 
 	if err := txBody(t.createQuery(tx)); err != nil {
-		return MapSQLError(err)
+		dbErr := MapSQLError(err)
+
+		// At this point, we know the DB wasn't able to properly
+		// serialize the error, so we'll re-execute everything to try
+		// once again.
+		var serializationErr *ErrSerializationError
+		if errors.As(dbErr, serializationErr) {
+			// Roll back the transaction, then pop back up to try
+			// once again.
+			//
+			// TODO(roasbeef): needs retry limit
```

Comment:
Should we fix this TODO to avoid endless re-tries?

Reply:
Yeah I think so, open question: what's a good limit for retries? My strategy was going to be deploy the modified version in staging/testnet, then check the logs to see how often we retry. We have @jharveyb's mega mint script, so that can be used to calibrate.